### ISSUE TITLE
Add support for parent names in addition to identifiers

### ISSUE THREAD

**Issue by sbraz** [scicore-unibas-ch/ansible-modules-guacamole#5]
Hi,
Would it be possible to allow specifying parents using names instead of identifiers? I think we could add a new `parent_name` parameter and translate it to `parent_identifier` in that function: https://github.com/scicore-unibas-ch/ansible-modules-guacamole/blob/ec8cebb491ebe29276bd10dfd710553019ab14b3/plugins/module_utils/guacamole.py#L60

**Comment by pescobar** [comment]
I don't know what's the difference between the parent name and the parent identifier. I thought it's the same. Can you provide more details?

**Comment by pescobar** [comment]
Is this what you mean? https://github.com/scicore-unibas-ch/ansible-modules-guacamole/pull/6/commits/1aba46afc91b59f909d4196fc7a90727dfa50c87

**Comment by sbraz** [comment]
> Is this what you mean? 1aba46a

Thanks but that is not what I mean.

> I thought it's the same

It might be dependent on your data source but with MySQL, group names are stored in the `name` attribute while the `identifier` is just an auto-incrementing integer. For instance, `/api/session/data/mysql/connectionGroups/` returns this on my setup:
```json
{

    "120": {
        "name": "GROUP_NAME1",
        "identifier": "120",
        "parentIdentifier": "ROOT",
        "type": "ORGANIZATIONAL",
        "activeConnections": 0,
        "attributes": {
            "max-connections": null,
            "max-connections-per-user": null,
            "enable-session-affinity": ""
        }
    },
    "121": {
        "name": "GROUP_NAME2",
        "identifier": "121",
        "parentIdentifier": "ROOT",
        "type": "ORGANIZATIONAL",
        "activeConnections": 0,
        "attributes": {
            "max-connections": null,
            "max-connections-per-user": null,
            "enable-session-affinity": ""
        }
    },
}
```

The idea would be to convert group names to group identifiers to make it easier to find them. Hope that makes sense, if not, let me know :)

**Comment by pescobar** [comment]
Ok I got it now. Thanks for the clarification. I did some tests with the api and I was not aware that the api expects the numeric identifier instead of the group name. I hadn't checked this before because I am not using groups.

Indeed this would be a nice addition for the module but it would require some extra additions:

  * do the mapping from group name to group numeric identifier
  * add the missing group if it doesn't exists
  * extend main() to check if new groups were added to report if something changed (idempotence)

I will try to create a PR to work on this new feature but I am quite busy now and I am not sure when I will be able to do it. Hopefully in coming days/weeks. If you have the time to work on it PRs are welcome ;)

**Comment by sbraz** [comment]
> do the mapping from group name to group numeric identifier

Yes this is the main thing I need :)

> add the missing group if it doesn't exists

This should be done by a separate module (`guacamole_group`?) that would take an additional argument that would be the parent group. The `guacamole_connection` module should not create groups on its own, it would make it too complex.

> check if new groups were added to report if something changed

I think all this will be handled by the future `guacamole_group` module too.

**Comment by pescobar** [comment]
I agree with you. You are right that adding a new module `guacamole_group` would be a better approach to keep the code simpler and easier to maintain. 

I will try this approach by first implementing the mapping from group name to group numeric identifier in the current `guacamole_connection` module when I find some time to work on it

**Comment by pescobar** [comment]
@sbraz I have updated PR https://github.com/scicore-unibas-ch/ansible-modules-guacamole/pull/6 and it should do what you need. I also took the opportunity to rename the argument `parentIdentifier` to `group_name`. You should be able to add connections to a custom group like this:

```
---

- name: Testing the Guacamole API
  hosts: localhost

  vars:
    local_guacamole_url: "http://localhost/guacamole"
    local_guacamole_user: "guacadmin"
    local_guacamole_password: "guacadmin"

  tasks:

    - name: Add a new RDP connection
      scicore.guacamole.guacamole_connection:
        base_url: "{{ local_guacamole_url }}"
        auth_username: "{{ local_guacamole_user }}"
        auth_password: "{{ local_guacamole_password }}"
        connection_name: test_connection_2
        groupName: "group1"
        protocol: rdp
        hostname: 192.168.33.44
        port: 3389
        username: rdp_user
        password: rdp_pass
      register: _connection_info

    - debug:
        var: _connection_info
```

I need to do some more testing before merging it to master. Can you check if it works fine for you?

**Comment by sbraz** [comment]
Thanks Pablo, we'll try this branch and let you know today or tomorrow.

**Comment by sbraz** [comment]
This works fine, thanks! Should I create a new issue to request a new `guacamole_group` module?

**Comment by pescobar** [comment]
I have released a new version `0.0.10` in https://galaxy.ansible.com/scicore/guacamole which includes the new feature to define `group_name`

I have started working in a new module `guacamole_connections_group` in https://github.com/scicore-unibas-ch/ansible-modules-guacamole/pull/7

**Comment by pescobar** [comment]
@sbraz module `guacamole_connections_group` in https://github.com/scicore-unibas-ch/ansible-modules-guacamole/pull/7 should be working. Can you test if it's working fine for you?
